### PR TITLE
[AC-2379] Stripe webhook update provider status

### PR DESCRIPTION
**Pull request by cturnbull-bitwarden**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Updates the Stripe webhook to enable or disable a given Provider if there are changes to the Provider's subscription status.

## Code changes

* **ProviderService.cs:** Two new methods, `EnableAsync` and `DisableAsync`, have been added. These methods change the Enabled property of a Provider and update the RevisionDate to the current UTC. They also handle updating the provider repository and application cache
* **Billing.csproj:** Updated to include a reference to `Commercial.Core.csproj` so that `IProviderService` can be added as a dependency
* **Startup.cs:** Added services to the DI container for IProviderService
* **StripeController:** Disabling or enabling a provider if its subscription status is either "unpaid" or "incomplete expired," or "active" respectively
* **Provider.cs:** Updated to allow the `RevisionDate` property to be set externally, enabling updates to this property from the new methods in `ProviderService.cs`
* **IProviderService.cs:** Updated to include declarations for the `EnableAsync` and `DisableAsync` methods
* **NoopProviderService.cs:** Dummy implementations of `EnableAsync` and `DisableAsync` have been added, throwing `NotImplementedException` when called

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by amorask-bitwarden — APPROVED** [review]

File: bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs [hunk 1]
```diff
@@ -647,6 +647,38 @@ public async Task DeleteAsync(Provider provider)
...
+    public async Task DisableAsync(Guid providerId)
+    {
+        var provider = await _providerRepository.GetByIdAsync(providerId);
+
+        if (provider is not { Enabled: true })
+        {
+            return;
+        }
+
+        provider.Enabled = true;
```
Comment (vincentsalucci) [inline bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs:675]: This should be `false`

**Review by vincentsalucci — CHANGES_REQUESTED** [review]

File: bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs (hunk 1 above)
Comment (cturnbull-bitwarden) [reply bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs]: Thanks for catching this! Just pushed an update

**Review by amorask-bitwarden — DISMISSED** [review]

**Review by vincentsalucci — DISMISSED** [review]
Non-blocking: only thing missing from this PR are a handful of tests for the new service methods added. Please follow-up at some point to get that coverage added.

**Review by amorask-bitwarden — APPROVED** [review]

**Review by vincentsalucci — APPROVED** [review]

**Comment by cturnbull-bitwarden** [comment]
Close this PR as we learned that this `Enabled` flag doesn't actually do anything. In the future, we'll just use the subscription status to decide if the Provider is enabled or not.

**Pull request closed without merging**